src/services/BaseService.js

```diff
@@ -35,7 +35,7 @@ function makeHttpRequest(method, uri, body, headers, callback) {
     }
   };
 
-  if (body !== undefined) {
+  if (typeof body === 'object') {
```

Comment:
This won't work because `null` is still an object. And besides, if it wasn't, the request wouldn't be sent at all because `request.send()` is inside this `if` statement.